From dc96c751ef2bb97fee404f7e394b1e90582b4f63 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 4 Apr 2016 21:18:28 -0400
Subject: init: Allow running dpkg-reconfigure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main.go | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/lxd/main.go b/lxd/main.go
index 2283e66..4a6613b 100644
--- a/lxd/main.go
+++ b/lxd/main.go
@@ -601,6 +601,7 @@ func cmdInit() error {
 	var networkPort int64     // Port
 	var trustPassword string  // Trust password
 	var imagesAutoUpdate bool // controls whether we set images.auto_update_interval to 0
+	var runReconfigure bool   // Whether to call dpkg-reconfigure
 
 	// Detect userns
 	defaultPrivileged = -1
@@ -738,6 +739,11 @@ func cmdInit() error {
 	}
 
 	if len(containers) > 0 || len(images) > 0 {
+		fmt.Printf(`LXD init cannot be used at this time.
+However if all you want to do is reconfigure the network,
+you can still do so by running "sudo dpkg-reconfigure -p medium lxd"
+
+`)
 		return fmt.Errorf("You have existing containers or images. lxd init requires an empty LXD.")
 	}
 
@@ -896,6 +902,10 @@ they otherwise would.
 		if !askBool("Would you like stale cached images to be updated automatically? (yes/no) [default=yes]? ", "yes") {
 			imagesAutoUpdate = false
 		}
+
+		if askBool("Do you want to configure the LXD bridge (yes/no) [default=yes]? ", "yes") {
+			runReconfigure = true
+		}
 	}
 
 	if !shared.StringInSlice(storageBackend, []string{"dir", "zfs"}) {
@@ -1001,6 +1011,17 @@ they otherwise would.
 		}
 	}
 
+	if runReconfigure {
+		cmd := exec.Command("dpkg-reconfigure", "-p", "medium", "lxd")
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("Failed to configure the bridge")
+		}
+	}
+
 	fmt.Printf("LXD has been successfully configured.\n")
 	return nil
 }
