From 8db6ab7d300cbe4256e9e8373e99dbb8771a66da Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 24 Feb 2016 13:41:45 -0500
Subject: Update lxc.mount.auto based on situation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/container_lxc.go | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/lxd/container_lxc.go b/lxd/container_lxc.go
index 9983067..ec2c87f 100644
--- a/lxd/container_lxc.go
+++ b/lxd/container_lxc.go
@@ -274,7 +274,21 @@ func (c *containerLXC) initLXC() error {
 		return err
 	}
 
-	err = lxcSetConfigItem(cc, "lxc.mount.auto", "cgroup:mixed proc:mixed sys:mixed")
+	// Set an appropriate /proc, /sys/ and /sys/fs/cgroup
+	mounts := []string{}
+	if c.IsPrivileged() && !runningInUserns {
+		mounts = append(mounts, "proc:mixed")
+		mounts = append(mounts, "sys:mixed")
+	} else {
+		mounts = append(mounts, "proc:rw")
+		mounts = append(mounts, "sys:rw")
+	}
+
+	if !shared.PathExists("/proc/self/ns/cgroup") {
+		mounts = append(mounts, "cgroup:mixed")
+	}
+
+	err = lxcSetConfigItem(cc, "lxc.mount.auto", strings.Join(mounts, " "))
 	if err != nil {
 		return err
 	}
