From ac1cb9d2383340e3739eb187755a3a268bdb6875 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 4 Apr 2016 21:18:28 -0400
Subject: init: Allow running dpkg-reconfigure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main_init.go | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/lxd/main_init.go b/lxd/main_init.go
index 76ddb1f..c85f952 100644
--- a/lxd/main_init.go
+++ b/lxd/main_init.go
@@ -26,6 +26,7 @@ func cmdInit() error {
 	var networkAddress string // Address
 	var networkPort int64     // Port
 	var trustPassword string  // Trust password
+	var runReconfigure bool   // Whether to call dpkg-reconfigure
 
 	// Detect userns
 	defaultPrivileged = -1
@@ -167,6 +168,11 @@ func cmdInit() error {
 	}
 
 	if len(containers) > 0 || len(images) > 0 {
+		fmt.Printf(`LXD init cannot be used at this time.
++However if all you want to do is reconfigure the network,
++you can still do so by running "sudo dpkg-reconfigure -p medium lxd"
+
+`)
 		return fmt.Errorf("You have existing containers or images. lxd init requires an empty LXD.")
 	}
 
@@ -321,6 +327,10 @@ they otherwise would.
 			networkPort = askInt("Port to bind LXD to [default=8443]: ", 1, 65535, "8443")
 			trustPassword = askPassword("Trust password for new clients: ")
 		}
+
+		if askBool("Do you want to configure the LXD bridge (yes/no) [default=yes]? ", "yes") {
+			runReconfigure = true
+		}
 	}
 
 	if !shared.StringInSlice(storageBackend, []string{"dir", "zfs"}) {
@@ -406,6 +416,17 @@ they otherwise would.
 		}
 	}
 
+	if runReconfigure {
+		cmd := exec.Command("dpkg-reconfigure", "-p", "medium", "lxd")
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("Failed to configure the bridge")
+		}
+	}
+
 	fmt.Printf("LXD has been successfully configured.\n")
 	return nil
 }
