From 6e3f879aa7abd765d5510ca5347367e74aaa1b56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 16 Feb 2017 13:12:24 -0500
Subject: tests: Fix mixed tab/spaces again
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/suites/storage.sh | 58 +++++++++++++++++++++++++-------------------------
 1 file changed, 29 insertions(+), 29 deletions(-)

diff --git a/test/suites/storage.sh b/test/suites/storage.sh
index 20d2c5e..e678a24 100644
--- a/test/suites/storage.sh
+++ b/test/suites/storage.sh
@@ -128,18 +128,18 @@ test_storage() {
     lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool1" "${BACKEND}"
 
     if [ "${BACKEND}" = "zfs" ]; then
-	# Let LXD use an already existing dataset.
-	zfs create -p -o mountpoint=none "lxdtest-$(basename "${LXD_DIR}")-pool1/existing-dataset-as-pool"
-	lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool7" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool1/existing-dataset-as-pool"
-
-	# Let LXD use an already existing storage pool.
-	configure_loop_device loop_file_4 loop_device_4
-	# shellcheck disable=SC2154
-	zpool create "lxdtest-$(basename "${LXD_DIR}")-pool9-existing-pool" "${loop_device_4}" -f -m none -O compression=on
-	lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool9" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool9-existing-pool"
-
-	# Let LXD create a new dataset and use as pool.
-	lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool8" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool1/non-existing-dataset-as-pool"
+      # Let LXD use an already existing dataset.
+      zfs create -p -o mountpoint=none "lxdtest-$(basename "${LXD_DIR}")-pool1/existing-dataset-as-pool"
+      lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool7" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool1/existing-dataset-as-pool"
+
+      # Let LXD use an already existing storage pool.
+      configure_loop_device loop_file_4 loop_device_4
+      # shellcheck disable=SC2154
+      zpool create "lxdtest-$(basename "${LXD_DIR}")-pool9-existing-pool" "${loop_device_4}" -f -m none -O compression=on
+      lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool9" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool9-existing-pool"
+
+      # Let LXD create a new dataset and use as pool.
+      lxc storage create "lxdtest-$(basename "${LXD_DIR}")-pool8" "${BACKEND}" source="lxdtest-$(basename "${LXD_DIR}")-pool1/non-existing-dataset-as-pool"
     fi
 
     # Create device backed zfs pool
@@ -201,14 +201,14 @@ test_storage() {
     lxc list -c b c12pool6 | grep "lxdtest-$(basename "${LXD_DIR}")-pool6"
 
     if [ "${BACKEND}" = "zfs" ]; then
-	lxc launch testimage c13pool7 -s "lxdtest-$(basename "${LXD_DIR}")-pool7"
-	lxc launch testimage c14pool7 -s "lxdtest-$(basename "${LXD_DIR}")-pool7"
+      lxc launch testimage c13pool7 -s "lxdtest-$(basename "${LXD_DIR}")-pool7"
+      lxc launch testimage c14pool7 -s "lxdtest-$(basename "${LXD_DIR}")-pool7"
 
-	lxc launch testimage c15pool8 -s "lxdtest-$(basename "${LXD_DIR}")-pool8"
-	lxc launch testimage c16pool8 -s "lxdtest-$(basename "${LXD_DIR}")-pool8"
+      lxc launch testimage c15pool8 -s "lxdtest-$(basename "${LXD_DIR}")-pool8"
+      lxc launch testimage c16pool8 -s "lxdtest-$(basename "${LXD_DIR}")-pool8"
 
-	lxc launch testimage c17pool9 -s "lxdtest-$(basename "${LXD_DIR}")-pool9"
-	lxc launch testimage c18pool9 -s "lxdtest-$(basename "${LXD_DIR}")-pool9"
+      lxc launch testimage c17pool9 -s "lxdtest-$(basename "${LXD_DIR}")-pool9"
+      lxc launch testimage c18pool9 -s "lxdtest-$(basename "${LXD_DIR}")-pool9"
     fi
 
     lxc delete -f c1pool1
@@ -230,24 +230,24 @@ test_storage() {
     lxc delete -f c12pool6
 
     if [ "${BACKEND}" = "zfs" ]; then
-	lxc delete -f c13pool7
-	lxc delete -f c14pool7
+      lxc delete -f c13pool7
+      lxc delete -f c14pool7
 
-	lxc delete -f c15pool8
-	lxc delete -f c16pool8
+      lxc delete -f c15pool8
+      lxc delete -f c16pool8
 
-	lxc delete -f c17pool9
-	lxc delete -f c18pool9
+      lxc delete -f c17pool9
+      lxc delete -f c18pool9
     fi
 
     lxc image delete testimage
 
     if [ "${BACKEND}" = "zfs" ]; then
-	lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool7"
-	lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool8"
-	lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool9"
-	# shellcheck disable=SC2154
-	deconfigure_loop_device "${loop_file_4}" "${loop_device_4}"
+      lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool7"
+      lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool8"
+      lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool9"
+      # shellcheck disable=SC2154
+      deconfigure_loop_device "${loop_file_4}" "${loop_device_4}"
     fi
 
     lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool2"
