From b97281c144379c7e92114231092220d100971d3b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Fri, 2 Mar 2018 22:56:43 -0500
Subject: tests: Try to avoid race condition in clustering tests
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/suites/clustering.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/suites/clustering.sh b/test/suites/clustering.sh
index b0122f3e..c97dc65f 100644
--- a/test/suites/clustering.sh
+++ b/test/suites/clustering.sh
@@ -103,6 +103,7 @@ test_clustering_membership() {
 
   # Remove a node gracefully.
   LXD_DIR="${LXD_FOUR_DIR}" lxc cluster delete node5
+  sleep 5
   LXD_DIR="${LXD_FOUR_DIR}" lxc cluster list | grep -q "https://0.0.0.0"
 
   LXD_DIR="${LXD_FOUR_DIR}" lxd shutdown
