From f0d06c8c75e10802a3fe2d6dccac07f37c69ff69 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 4 Apr 2016 21:18:28 -0400
Subject: init: Allow running dpkg-reconfigure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main.go | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/lxd/main.go b/lxd/main.go
index 9f90613..f459ed7 100644
--- a/lxd/main.go
+++ b/lxd/main.go
@@ -583,6 +583,7 @@ func cmdInit() error {
 	var networkAddress string // Address
 	var networkPort int       // Port
 	var trustPassword string  // Trust password
+	var runReconfigure bool   // Whether to call dpkg-reconfigure
 
 	// Only root should run this
 	if os.Geteuid() != 0 {
@@ -790,6 +791,10 @@ func cmdInit() error {
 			networkPort = askInt("Port to bind LXD to (8443 recommended): ", 1, 65535)
 			trustPassword = askPassword("Trust password for new clients: ")
 		}
+
+		if askBool("Do you want to configure the LXD bridge (yes/no)? ") {
+			runReconfigure = true
+		}
 	}
 
 	if !shared.StringInSlice(storageBackend, []string{"dir", "zfs"}) {
@@ -861,6 +866,17 @@ func cmdInit() error {
 		}
 	}
 
+	if runReconfigure {
+		cmd := exec.Command("dpkg-reconfigure", "-p", "medium", "lxd")
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("Failed to configure the bridge")
+		}
+	}
+
 	fmt.Printf("LXD has been successfully configured.\n")
 	return nil
 }
