From 9140d6251ecebd22fab784867d0c31511509410d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 6 Mar 2018 23:14:48 +0100
Subject: lxd/shutdown: Fix error string check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main_shutdown.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxd/main_shutdown.go b/lxd/main_shutdown.go
index 3de15e3e..53f80366 100644
--- a/lxd/main_shutdown.go
+++ b/lxd/main_shutdown.go
@@ -2,7 +2,7 @@ package main
 
 import (
 	"fmt"
-	"io"
+	"strings"
 	"time"
 
 	"github.com/lxc/lxd/client"
@@ -18,7 +18,7 @@ func cmdShutdown(args *Args) error {
 	}
 
 	_, _, err = c.RawQuery("PUT", "/internal/shutdown", nil, "")
-	if err != nil && err != io.EOF {
+	if err != nil && strings.HasSuffix(err.Error(), ": EOF") {
 		// NOTE: if we got an EOF error here it means that the daemon
 		// has shutdown so quickly that it already closed the unix
 		// socket. We consider the daemon dead in this case.
