From 5b259e3fcb914db70c68a2381ef11bbef6637382 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 21 Dec 2016 03:26:54 -0500
Subject: tests: Fix shellcheck being confused by cd
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/suites/filemanip.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/suites/filemanip.sh b/test/suites/filemanip.sh
index 9f970e3..ab7ade6 100644
--- a/test/suites/filemanip.sh
+++ b/test/suites/filemanip.sh
@@ -1,6 +1,8 @@
 #!/bin/sh
 
 test_filemanip() {
+  # Workaround for shellcheck getting confused by "cd"
+  set -e
   ensure_import_testimage
   ensure_has_localhost_remote "${LXD_ADDR}"
 
