From 7adf74fd5f2c4699ced7b5b2bbb7e5fa54cce941 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 24 Aug 2017 01:17:33 -0400
Subject: network: Make dnsmasq quiet when not in debug mode
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/networks.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lxd/networks.go b/lxd/networks.go
index 79267cbc..0780967f 100644
--- a/lxd/networks.go
+++ b/lxd/networks.go
@@ -769,6 +769,10 @@ func (n *network) Start() error {
 		"--except-interface=lo",
 		fmt.Sprintf("--interface=%s", n.name)}
 
+	if !debug {
+		dnsmasqCmd = append(dnsmasqCmd, []string{"--quiet-dhcp", "--quiet-dhcp6", "--quiet-ra"}...)
+	}
+
 	// Configure IPv4
 	if !shared.StringInSlice(n.config["ipv4.address"], []string{"", "none"}) {
 		// Parse the subnet
