From 17573d54c4ae445b598f9e8cbf7d9a524ede2756 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 28 Sep 2016 22:57:57 -0400
Subject: init: Actually add new network to profile
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lxd/main.go b/lxd/main.go
index 56ef732..c3b491a 100644
--- a/lxd/main.go
+++ b/lxd/main.go
@@ -1028,6 +1028,12 @@ they otherwise would.
 		if err != nil {
 			return err
 		}
+
+		props := []string{"nictype=bridged", fmt.Sprintf("parent=%s", bridgeName)}
+		_, err = c.ProfileDeviceAdd("default", "eth0", "nic", props)
+		if err != nil {
+			return err
+		}
 	}
 
 	fmt.Printf("LXD has been successfully configured.\n")
