From f6f8ae94af86ff50971e6b4c54144fba558f5089 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 5 Dec 2016 11:25:45 +0100
Subject: tests: Test lxd shutdown
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/suites/basic.sh | 25 +++++++++++++++++++++++--
 1 file changed, 23 insertions(+), 2 deletions(-)

diff --git a/test/suites/basic.sh b/test/suites/basic.sh
index 9dcd9d3..a0f4674 100644
--- a/test/suites/basic.sh
+++ b/test/suites/basic.sh
@@ -215,8 +215,9 @@ test_basic_usage() {
   [ "$(my_curl "https://${LXD_ADDR}/1.0/containers/configtest" | jq -r .metadata.config[\"raw.lxc\"])" = "lxc.hook.clone=/bin/true" ]
   lxc delete configtest
 
-  # Test socket activation
+  # Test activateifneeded/shutdown
   LXD_ACTIVATION_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_ACTIVATION_DIR}"
   spawn_lxd "${LXD_ACTIVATION_DIR}"
   (
     set -e
@@ -230,7 +231,27 @@ test_basic_usage() {
     lxd activateifneeded --debug 2>&1 | grep -q -v "activating..."
     lxc config set autostart boot.autostart true --force-local
     lxd activateifneeded --debug 2>&1 | grep -q "Daemon has auto-started containers, activating..."
-    lxc delete autostart --force-local
+
+    lxc config unset autostart boot.autostart --force-local
+    lxd activateifneeded --debug 2>&1 | grep -q -v "activating..."
+
+    lxc start autostart --force-local
+    PID=$(lxc info autostart --force-local | grep ^Pid | awk '{print $2}')
+    lxd shutdown
+    [ -d "/proc/${PID}" ] && false
+
+    lxd activateifneeded --debug 2>&1 | grep -q "Daemon has auto-started containers, activating..."
+
+    # shellcheck disable=SC2086
+    lxd --logfile "${LXD_DIR}/lxd.log" ${DEBUG-} "$@" 2>&1 &
+    LXD_PID=$!
+    echo "${LXD_PID}" > "${LXD_DIR}/lxd.pid"
+    echo "${LXD_DIR}" >> "${TEST_DIR}/daemons"
+    lxd waitready --timeout=300
+
+    lxc list --force-local autostart | grep -q RUNNING
+
+    lxc delete autostart --force --force-local
   )
   # shellcheck disable=SC2031
   LXD_DIR=${LXD_DIR}
