From 76ab4208745543090e71512938190c6fdbba6dbc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 30 Jun 2016 13:30:28 -0400
Subject: simplestreams: list images available as both squashfs and tar.xz
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 shared/simplestreams.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shared/simplestreams.go b/shared/simplestreams.go
index a3d07c1..973b9bb 100644
--- a/shared/simplestreams.go
+++ b/shared/simplestreams.go
@@ -153,7 +153,7 @@ func (s *SimpleStreamsManifest) ToLXD() ([]ImageInfo, map[string][][]string) {
 				}
 			}
 
-			if found != 2 || size == 0 || filename == "" || fingerprint == "" {
+			if found < 2 || size == 0 || filename == "" || fingerprint == "" {
 				// Invalid image
 				continue
 			}
