From 9752c2d373b3bd00cd0f4d427a8e68c568772530 Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho.andersen@canonical.com>
Date: Thu, 7 Jan 2016 08:50:27 -0700
Subject: lxc copy: try and report the real error

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>
---
 lxc/copy.go | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/lxc/copy.go b/lxc/copy.go
index 47b51a8..a07480e 100644
--- a/lxc/copy.go
+++ b/lxc/copy.go
@@ -138,28 +138,50 @@ func copyContainer(config *lxd.Config, sourceResource string, destResource strin
 			return err
 		}
 
+		/* Since we're trying a bunch of different network ports that
+		 * may be invalid, we can get "bad handshake" errors when the
+		 * websocket code tries to connect. If the first error is a
+		 * real error, but the subsequent errors are only network
+		 * errors, we should try to report the first real error. Of
+		 * course, if all the errors are websocket errors, let's just
+		 * report that.
+		 */
+		var realError error
+
 		for _, addr := range addresses {
 			var migration *lxd.Response
 
 			sourceWSUrl := "https://" + addr + sourceWSResponse.Operation
 			migration, err = dest.MigrateFrom(destName, sourceWSUrl, secrets, status.Architecture, status.Config, status.Devices, status.Profiles, baseImage, ephemeral == 1)
 			if err != nil {
+				if !strings.Contains(err.Error(), "websocket: bad handshake") {
+					realError = err
+				}
 				shared.Debugf("intermediate error: %s", err)
 				continue
 			}
 
 			if err = dest.WaitForSuccess(migration.Operation); err != nil {
+				if !strings.Contains(err.Error(), "websocket: bad handshake") {
+					realError = err
+				}
 				shared.Debugf("intermediate error: %s", err)
 				// FIXME: This is a backward compatibility codepath
 				sourceWSUrl := "wss://" + addr + sourceWSResponse.Operation + "/websocket"
 
 				migration, err = dest.MigrateFrom(destName, sourceWSUrl, secrets, status.Architecture, status.Config, status.Devices, status.Profiles, baseImage, ephemeral == 1)
 				if err != nil {
+					if !strings.Contains(err.Error(), "websocket: bad handshake") {
+						realError = err
+					}
 					shared.Debugf("intermediate error: %s", err)
 					continue
 				}
 
 				if err = dest.WaitForSuccess(migration.Operation); err != nil {
+					if !strings.Contains(err.Error(), "websocket: bad handshake") {
+						realError = err
+					}
 					shared.Debugf("intermediate error: %s", err)
 					continue
 				}
@@ -168,7 +190,11 @@ func copyContainer(config *lxd.Config, sourceResource string, destResource strin
 			return nil
 		}
 
-		return err
+		if realError != nil {
+			return realError
+		} else {
+			return err
+		}
 	}
 }
 
