From b91ad39db983ec87e0032b6e2faa56dea4f88dcc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 5 Dec 2016 11:45:47 +0100
Subject: Only mark ready once containers are up
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/daemon.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/daemon.go b/lxd/daemon.go
index 23d2a38..eb97e11 100644
--- a/lxd/daemon.go
+++ b/lxd/daemon.go
@@ -1062,7 +1062,7 @@ func (d *Daemon) Ready() error {
 	}()
 
 	/* Restore containers */
-	go containersRestart(d)
+	containersRestart(d)
 
 	/* Re-balance in case things changed while LXD was down */
 	deviceTaskBalance(d)
