From 710c5eeef1d3a4fc630f4132074a5e6a0e5f67c5 Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho.andersen@canonical.com>
Date: Thu, 25 Feb 2016 09:22:30 -0700
Subject: exec: don't s/\\n/\\r\\n/g when interactive

Closes #1652 (again :)

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>
---
 shared/util_linux.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/shared/util_linux.go b/shared/util_linux.go
index d9e7bc1..279d67c 100644
--- a/shared/util_linux.go
+++ b/shared/util_linux.go
@@ -59,6 +59,8 @@ void configure_pty(int fd) {
 
 	term_settings.c_cflag |= HUPCL;
 
+	term_settings.c_oflag &= ~ONLCR;
+
 	if (tcsetattr(fd, TCSANOW, &term_settings) < 0) {
 		fprintf(stderr, "Failed to set settings: %s\n", strerror(errno));
 		return;
