From 04e2b2ba2871dfc72d3b617387abd5b57a7fc562 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 24 Mar 2017 13:24:11 +0100
Subject: containers_post: better error message

Closes #3110.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/containers_post.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lxd/containers_post.go b/lxd/containers_post.go
index 2db5f179..1fce235a 100644
--- a/lxd/containers_post.go
+++ b/lxd/containers_post.go
@@ -223,10 +223,14 @@ func createFromMigration(d *Daemon, req *api.ContainersPost) Response {
 		}
 	}
 
+	shared.LogDebugf("No valid storage pool in the container's local root disk device and profiles found.")
 	// If there is just a single pool in the database, use that
 	if storagePool == "" {
 		pools, err := dbStoragePools(d.db)
 		if err != nil {
+			if err == NoSuchObjectError {
+				return BadRequest(fmt.Errorf("This LXD instance does not have any storage pools configured."))
+			}
 			return InternalError(err)
 		}
 
