From a06c459de69ec43ecc696b985566e3a7e4b48c7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 21 Aug 2017 16:12:38 -0400
Subject: lxd/images: Carry old "cached" value on refresh
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #3698

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/images.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lxd/images.go b/lxd/images.go
index dcda5560..298c0caf 100644
--- a/lxd/images.go
+++ b/lxd/images.go
@@ -899,6 +899,14 @@ func autoUpdateImage(d *Daemon, op *operation, fingerprint string, id int, info
 		return err
 	}
 
+	if info.Cached {
+		err = dbImageLastAccessInit(d.db, hash)
+		if err != nil {
+			logger.Error("Error setting cached flag", log.Ctx{"err": err, "fp": hash})
+			return err
+		}
+	}
+
 	err = dbImageLastAccessUpdate(d.db, hash, info.LastUsedAt)
 	if err != nil {
 		logger.Error("Error setting last use date", log.Ctx{"err": err, "fp": hash})
