From 5aafd7a08040fc3eaee2b1205fdf83b6b2519744 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 13 Jun 2017 16:02:18 -0400
Subject: btrfs: Apply default flags BEFORE detecting type
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #3409

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/storage_btrfs.go | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/lxd/storage_btrfs.go b/lxd/storage_btrfs.go
index a4ea3997..d7910b92 100644
--- a/lxd/storage_btrfs.go
+++ b/lxd/storage_btrfs.go
@@ -361,7 +361,6 @@ func (s *storageBtrfs) StoragePoolMount() (bool, error) {
 		}
 		lxdStorageMapLock.Unlock()
 	}
-
 	defer removeLockFromMap()
 
 	// Check whether the mount poolMntPoint exits.
@@ -376,7 +375,7 @@ func (s *storageBtrfs) StoragePoolMount() (bool, error) {
 		return false, nil
 	}
 
-	mountFlags := uintptr(0)
+	mountFlags, mountOptions := lxdResolveMountoptions(s.getBtrfsMountOptions())
 	mountSource := source
 	isBlockDev := shared.IsBlockdevPath(source)
 	if filepath.IsAbs(source) {
@@ -398,7 +397,7 @@ func (s *storageBtrfs) StoragePoolMount() (bool, error) {
 			defer loopF.Close()
 		} else if !isBlockDev && cleanSource != poolMntPoint {
 			mountSource = source
-			mountFlags = syscall.MS_BIND
+			mountFlags |= syscall.MS_BIND
 		} else if !isBlockDev && cleanSource == poolMntPoint && s.d.BackingFs == "btrfs" {
 			return false, nil
 		}
@@ -419,14 +418,12 @@ func (s *storageBtrfs) StoragePoolMount() (bool, error) {
 			// detection task.
 			return false, nil
 		}
-
 	}
 
-	mountFlags, mountOptions := lxdResolveMountoptions(s.getBtrfsMountOptions())
 	mountFlags |= s.remount
 	err := syscall.Mount(mountSource, poolMntPoint, "btrfs", mountFlags, mountOptions)
 	if err != nil {
-		logger.Errorf("failed to mount BTRFS storage pool \"%s\" onto \"%s\" with mountoptions \"%s\": %s", mountSource, poolMntPoint, mountOptions, err)
+		logger.Errorf("Failed to mount BTRFS storage pool \"%s\" onto \"%s\" with mountoptions \"%s\": %s", mountSource, poolMntPoint, mountOptions, err)
 		return false, err
 	}
 
