From 3e5573fd12d26fefd917eae5624ff35558e04460 Mon Sep 17 00:00:00 2001
From: Thomas Johnson <NTmatter@gmail.com>
Date: Mon, 27 Mar 2017 18:34:12 +0200
Subject: lxd/container: Typo in securtiy.syscalls.blacklist

Signed-off-by: Thomas Johnson <NTmatter@gmail.com>
---
 lxd/container.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/container.go b/lxd/container.go
index 230fe863..4dd72f8b 100644
--- a/lxd/container.go
+++ b/lxd/container.go
@@ -208,7 +208,7 @@ func containerValidConfig(d *Daemon, config map[string]string, profile bool, exp
 
 	_, rawSeccomp := config["raw.seccomp"]
 	_, whitelist := config["security.syscalls.whitelist"]
-	_, blacklist := config["securtiy.syscalls.blacklist"]
+	_, blacklist := config["security.syscalls.blacklist"]
 	blacklistDefault := shared.IsTrue(config["security.syscalls.blacklist_default"])
 	blacklistCompat := shared.IsTrue(config["security.syscalls.blacklist_compat"])
 
