From 41b820a346ca42597ff2c0ac812e8a4011f21f6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 28 Sep 2016 02:14:27 -0400
Subject: Fix typo discovered by lintian
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/networks_utils.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/networks_utils.go b/lxd/networks_utils.go
index f68f20c..43342b7 100644
--- a/lxd/networks_utils.go
+++ b/lxd/networks_utils.go
@@ -568,7 +568,7 @@ func networkFanAddress(underlay *net.IPNet, overlay *net.IPNet) (string, string,
 	}
 
 	if overlaySize+(32-underlaySize)+8 > 32 {
-		return "", "", "", fmt.Errorf("Underlay or overlay networks too large to accomodate the FAN")
+		return "", "", "", fmt.Errorf("Underlay or overlay networks too large to accommodate the FAN")
 	}
 
 	// Get the IP
