From a5e3d80ec42840dc536f81e27ed34632bddce3ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 20 Jun 2017 11:18:14 -0400
Subject: lxd/images: Initialize image info in direct case
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/daemon_images.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lxd/daemon_images.go b/lxd/daemon_images.go
index b2023d42..33d02395 100644
--- a/lxd/daemon_images.go
+++ b/lxd/daemon_images.go
@@ -466,6 +466,8 @@ func (d *Daemon) ImageDownload(op *operation, server string, protocol string, ce
 			return nil, err
 		}
 
+		info = &api.Image{}
+		info.Fingerprint = fp
 		info.Size = size
 		info.Architecture = imageMeta.Architecture
 		info.CreatedAt = time.Unix(imageMeta.CreationDate, 0)
