From 5544f7956073d38457371f0fcc85c0c404d91a11 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 2 Jan 2018 18:47:15 +0100
Subject: lxd/init: Strip leading and trailing spaces
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes: #4125

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 shared/cmd/context.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/shared/cmd/context.go b/shared/cmd/context.go
index 0caa946b..251a5240 100644
--- a/shared/cmd/context.go
+++ b/shared/cmd/context.go
@@ -161,6 +161,7 @@ func (c *Context) invalidInput() {
 func (c *Context) readAnswer(defaultAnswer string) string {
 	answer, _ := c.stdin.ReadString('\n')
 	answer = strings.TrimSuffix(answer, "\n")
+	answer = strings.TrimSpace(answer)
 	if answer == "" {
 		answer = defaultAnswer
 	}
