From b02e491d7c6428b1c0c4619a76f0f25c68907c1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 28 Mar 2017 17:04:44 -0400
Subject: operations: Remove useless for loops
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/operations.go | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/lxd/operations.go b/lxd/operations.go
index f285d051..517652b1 100644
--- a/lxd/operations.go
+++ b/lxd/operations.go
@@ -283,23 +283,19 @@ func (op *operation) WaitFinal(timeout int) (bool, error) {
 
 	// Wait indefinitely
 	if timeout == -1 {
-		for {
-			<-op.chanDone
-			return true, nil
-		}
+		<-op.chanDone
+		return true, nil
 	}
 
 	// Wait until timeout
 	if timeout > 0 {
 		timer := time.NewTimer(time.Duration(timeout) * time.Second)
-		for {
-			select {
-			case <-op.chanDone:
-				return false, nil
+		select {
+		case <-op.chanDone:
+			return false, nil
 
-			case <-timer.C:
-				return false, nil
-			}
+		case <-timer.C:
+			return false, nil
 		}
 	}
 
