From b0a52db5920c9539d330616a3cf688a44acc749a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sun, 21 Feb 2016 00:42:40 -0500
Subject: Use iproute2 instead of bridge-utils
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/container_lxc.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/container_lxc.go b/lxd/container_lxc.go
index c8ca8c4..0701841 100644
--- a/lxd/container_lxc.go
+++ b/lxd/container_lxc.go
@@ -3100,7 +3100,7 @@ func (c *containerLXC) createNetworkDevice(name string, m shared.Device) (string
 		}
 
 		if m["nictype"] == "bridged" {
-			err = exec.Command("brctl", "addif", m["parent"], n1).Run()
+			err = exec.Command("ip", "link", "set", n1, "master", m["parent"]).Run()
 			if err != nil {
 				deviceRemoveInterface(n2)
 				return "", fmt.Errorf("Failed to add interface to bridge: %s", err)
