From 730e93c206076da9089c81c55d0c93fda933c8d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 2 Apr 2018 01:11:43 -0400
Subject: lxc/image: Fix crash due to bad arg parsing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxc/image.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lxc/image.go b/lxc/image.go
index 26e633ef..7baac50d 100644
--- a/lxc/image.go
+++ b/lxc/image.go
@@ -642,7 +642,6 @@ func (c *cmdImageImport) Run(cmd *cobra.Command, args []string) error {
 
 	if imageFile == "" {
 		imageFile = args[0]
-		properties = properties[1:]
 	}
 
 	if shared.PathExists(shared.HostPath(filepath.Clean(imageFile))) {
