From 5593a18a3a71316f8bb001b0390289c3eb1c24f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sat, 23 Sep 2017 22:28:49 -0400
Subject: version: Only include kernel version, not build id
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 shared/version/platform_linux.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/shared/version/platform_linux.go b/shared/version/platform_linux.go
index 55deeadc..1be9b342 100644
--- a/shared/version/platform_linux.go
+++ b/shared/version/platform_linux.go
@@ -3,6 +3,8 @@
 package version
 
 import (
+	"strings"
+
 	"github.com/lxc/lxd/shared"
 	"github.com/lxc/lxd/shared/osarch"
 )
@@ -13,7 +15,7 @@ func getPlatformVersionStrings() []string {
 	// add kernel version
 	uname, err := shared.Uname()
 	if err == nil {
-		versions = append(versions, uname.Release)
+		versions = append(versions, strings.Split(uname.Release, "-")[0])
 	}
 
 	// add distribution info
