From 5c943d913b0aa146e1ed7ea2d19c46b4ad85e4fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Fri, 30 Sep 2016 23:00:39 -0400
Subject: network: Properly catch ENOENT
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/networks.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxd/networks.go b/lxd/networks.go
index 6a056b4..9988c3e 100644
--- a/lxd/networks.go
+++ b/lxd/networks.go
@@ -841,7 +841,7 @@ func (n *network) Start() error {
 				}
 
 				err = networkSysctl(fmt.Sprintf("ipv6/conf/%s/accept_ra", entry.Name()), "2")
-				if err != nil && err != os.ErrNotExist {
+				if err != nil && !os.IsNotExist(err) {
 					return err
 				}
 			}
@@ -849,7 +849,7 @@ func (n *network) Start() error {
 			// Then set forwarding for all of them
 			for _, entry := range entries {
 				err = networkSysctl(fmt.Sprintf("ipv6/conf/%s/forwarding", entry.Name()), "1")
-				if err != nil && err != os.ErrNotExist {
+				if err != nil && !os.IsNotExist(err) {
 					return err
 				}
 			}
