From 212e4ffcc6982895907628e845a1f2660cecb1c3 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 5 Apr 2017 18:30:18 +0200
Subject: client: fix file push

Closes #3153.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 client.go | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/client.go b/client.go
index 520228e9..535e6afd 100644
--- a/client.go
+++ b/client.go
@@ -1832,7 +1832,10 @@ func (c *Client) MkdirP(container string, p string, mode os.FileMode, uid int, g
 		return nil
 	}
 
-	parts := strings.Split(p, "/")
+	// Remove trailing "/" e.g. /A/B/C/. Otherwise we will end up with an
+	// empty array entry "" which will confuse the Mkdir() loop below.
+	pclean := filepath.Clean(p)
+	parts := strings.Split(pclean, "/")
 	i := len(parts)
 
 	for ; i >= 1; i-- {
@@ -1852,6 +1855,10 @@ func (c *Client) MkdirP(container string, p string, mode os.FileMode, uid int, g
 
 	for ; i <= len(parts); i++ {
 		cur := filepath.Join(parts[:i]...)
+		if cur == "" {
+			continue
+		}
+
 		if err := c.Mkdir(container, cur, mode, uid, gid); err != nil {
 			return err
 		}
