From 4dd6bd76702ead01b5973b9eca3f8a96866e2e93 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 9 Aug 2017 17:34:23 +0200
Subject: devices: handle empty isolcpus set

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/devices.go | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/lxd/devices.go b/lxd/devices.go
index b35a198d..a140e8a3 100644
--- a/lxd/devices.go
+++ b/lxd/devices.go
@@ -523,16 +523,20 @@ func deviceTaskBalance(d *Daemon) {
 	isolatedCpusInt := effectiveCpusInt
 
 	if shared.PathExists("/sys/devices/system/cpu/isolated") {
-		isolatedCpus, err := ioutil.ReadFile("/sys/devices/system/cpu/isolated")
+		buf, err := ioutil.ReadFile("/sys/devices/system/cpu/isolated")
 		if err != nil {
 			logger.Errorf("Error reading host's isolated cpu")
 			return
 		}
 
-		isolatedCpusInt, err = parseCpuset(strings.TrimRight(string(isolatedCpus), "\n"))
-		if err != nil {
-			logger.Errorf("Error parsing isolated CPU set: %s", string(isolatedCpus))
-			return
+		// File might exist even though there are no isolated cpus.
+		isolatedCpus := strings.TrimSpace(string(buf))
+		if isolatedCpus != "" {
+			isolatedCpusInt, err = parseCpuset(isolatedCpus)
+			if err != nil {
+				logger.Errorf("Error parsing isolated CPU set: %s", string(isolatedCpus))
+				return
+			}
 		}
 	}
 
