From 2fe5e84d0677ce98da20325108a9d7ca34a03d78 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 9 Aug 2017 19:55:46 +0200
Subject: devices: do not mark all cpus isolated by default

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/devices.go | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lxd/devices.go b/lxd/devices.go
index a140e8a3..771094b5 100644
--- a/lxd/devices.go
+++ b/lxd/devices.go
@@ -520,8 +520,7 @@ func deviceTaskBalance(d *Daemon) {
 		return
 	}
 
-	isolatedCpusInt := effectiveCpusInt
-
+	isolatedCpusInt := []int{}
 	if shared.PathExists("/sys/devices/system/cpu/isolated") {
 		buf, err := ioutil.ReadFile("/sys/devices/system/cpu/isolated")
 		if err != nil {
