From b7b0048a12e56560255373d4acb5b73a0c540f2a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 8 Jan 2018 21:32:20 -0500
Subject: storage: Serialize storage pool creation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #4150

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/storage_pools.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lxd/storage_pools.go b/lxd/storage_pools.go
index 6d4d80bc..a676cdca 100644
--- a/lxd/storage_pools.go
+++ b/lxd/storage_pools.go
@@ -6,6 +6,7 @@ import (
 	"net/http"
 	"strconv"
 	"strings"
+	"sync"
 
 	"github.com/gorilla/mux"
 	"github.com/lxc/lxd/lxd/db"
@@ -14,6 +15,9 @@ import (
 	"github.com/lxc/lxd/shared/version"
 )
 
+// Lock to prevent concurent storage pools creation
+var storagePoolCreateLock sync.Mutex
+
 // /1.0/storage-pools
 // List all storage pools.
 func storagePoolsGet(d *Daemon, r *http.Request) Response {
@@ -60,6 +64,9 @@ func storagePoolsGet(d *Daemon, r *http.Request) Response {
 // /1.0/storage-pools
 // Create a storage pool.
 func storagePoolsPost(d *Daemon, r *http.Request) Response {
+	storagePoolCreateLock.Lock()
+	defer storagePoolCreateLock.Unlock()
+
 	req := api.StoragePoolsPost{}
 
 	// Parse the request.
