From c6f5b000fa650614ddaeaf31c294416d643434f7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 9 Jan 2018 00:51:30 -0500
Subject: lxd/containers: Fix container shutdown on exit
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #4130

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/containers.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxd/containers.go b/lxd/containers.go
index 6a27d0fa..800ff5f2 100644
--- a/lxd/containers.go
+++ b/lxd/containers.go
@@ -236,13 +236,13 @@ func containersShutdown(s *state.State) error {
 
 			// Stop the container
 			wg.Add(1)
-			go func() {
+			go func(c container, lastState string) {
 				c.Shutdown(time.Second * time.Duration(timeoutSeconds))
 				c.Stop(false)
 				c.ConfigKeySet("volatile.last_state.power", lastState)
 
 				wg.Done()
-			}()
+			}(c, lastState)
 		} else {
 			c.ConfigKeySet("volatile.last_state.power", lastState)
 		}
