From f905f31c75bb3a47caf70bccdd835a4f79218c11 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 5 Oct 2017 17:21:22 -0400
Subject: images: Respect disabled cache expiry
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/images.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lxd/images.go b/lxd/images.go
index 973ed850..c7ebdbca 100644
--- a/lxd/images.go
+++ b/lxd/images.go
@@ -1010,6 +1010,10 @@ func pruneExpiredImages(d *Daemon) {
 
 	// Get the list of expired images.
 	expiry := daemonConfig["images.remote_cache_expiry"].GetInt64()
+	if expiry <= 0 {
+		return
+	}
+
 	images, err := db.ImagesGetExpired(d.db, expiry)
 	if err != nil {
 		logger.Error("Unable to retrieve the list of expired images", log.Ctx{"err": err})
