From 87f3a0bd3201479d2064e11d23f9fe94039b1dea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 5 Oct 2017 17:48:24 -0400
Subject: patches: Convert UploadedAt to RFC3399
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/patches.go | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/lxd/patches.go b/lxd/patches.go
index 2c630b10..6f2c8977 100644
--- a/lxd/patches.go
+++ b/lxd/patches.go
@@ -49,6 +49,7 @@ var patches = []patch{
 	{name: "storage_zfs_volume_size", run: patchStorageZFSVolumeSize},
 	{name: "network_dnsmasq_hosts", run: patchNetworkDnsmasqHosts},
 	{name: "storage_api_dir_bind_mount", run: patchStorageApiDirBindMount},
+	{name: "fix_uploaded_at", run: patchFixUploadedAt},
 }
 
 type patch struct {
@@ -2475,6 +2476,27 @@ func patchStorageApiDirBindMount(name string, d *Daemon) error {
 	return nil
 }
 
+func patchFixUploadedAt(name string, d *Daemon) error {
+	images, err := db.ImagesGet(d.db, false)
+	if err != nil {
+		return err
+	}
+
+	for _, fingerprint := range images {
+		id, image, err := db.ImageGet(d.db, fingerprint, false, true)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(d.db, "UPDATE images SET upload_date=? WHERE id=?", image.UploadedAt, id)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 // Patches end here
 
 // Here are a couple of legacy patches that were originally in
