From 738db8ce606e9c3bbdb3282f8810348c2315ae78 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 3 Apr 2018 20:30:41 -0400
Subject: lxd/init: Don't setup a remote storage pool by default
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/main_init_interactive.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/main_init_interactive.go b/lxd/main_init_interactive.go
index 994f472b..89863201 100644
--- a/lxd/main_init_interactive.go
+++ b/lxd/main_init_interactive.go
@@ -375,7 +375,7 @@ func (c *cmdInit) askStorage(config *initData, d lxd.ContainerServer) error {
 			}
 		}
 
-		if cli.AskBool("Do you want to configure a new remote storage pool (yes/no) [default=yes]? ", "yes") {
+		if cli.AskBool("Do you want to configure a new remote storage pool (yes/no) [default=no]? ", "no") {
 			err := c.askStoragePool(config, d, "remote")
 			if err != nil {
 				return err
