From b932df0b93e546c4f3bafd6b3d9ce2c97db7efc1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 10 Jan 2018 22:16:29 -0500
Subject: lxc/file: Fix edit in a snap environment
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxc/file.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxc/file.go b/lxc/file.go
index 545982d9..d0b5cb12 100644
--- a/lxc/file.go
+++ b/lxc/file.go
@@ -577,7 +577,7 @@ func (c *fileCmd) edit(conf *config.Config, args []string) error {
 	fname := f.Name()
 	f.Close()
 	os.Remove(fname)
-	defer os.Remove(fname)
+	defer os.Remove(shared.HostPath(fname))
 
 	// Extract current value
 	err = c.pull(conf, append([]string{args[0]}, fname))
@@ -585,7 +585,7 @@ func (c *fileCmd) edit(conf *config.Config, args []string) error {
 		return err
 	}
 
-	_, err = shared.TextEditor(fname, []byte{})
+	_, err = shared.TextEditor(shared.HostPath(fname), []byte{})
 	if err != nil {
 		return err
 	}
