From 9266eb4569bec762a35b5b87ae9ed64170552172 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 5 Oct 2017 18:43:50 -0400
Subject: daemon: Don't update images while pruning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/daemon.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lxd/daemon.go b/lxd/daemon.go
index 8b41fd58..4788b698 100644
--- a/lxd/daemon.go
+++ b/lxd/daemon.go
@@ -797,7 +797,6 @@ func (d *Daemon) Ready() error {
 	/* Prune images */
 	d.pruneChan = make(chan bool)
 	go func() {
-		pruneExpiredImages(d)
 		for {
 			timer := time.NewTimer(24 * time.Hour)
 			timeChan := timer.C
@@ -813,6 +812,9 @@ func (d *Daemon) Ready() error {
 		}
 	}()
 
+	// Do an initial pruning run before we start updating images
+	pruneExpiredImages(d)
+
 	/* Auto-update images */
 	d.resetAutoUpdateChan = make(chan bool)
 	go func() {
