From 33143f7a7f0a2806e4897fde2d10080456220c3a Mon Sep 17 00:00:00 2001
From: David Negreira <david.negreira@emesa.nl>
Date: Fri, 12 Jan 2018 22:33:52 +0100
Subject: lxc client: return errArgs on empty pool

Signed-off-by: David Negreira <David@otherreality.net>
---
 lxc/storage.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lxc/storage.go b/lxc/storage.go
index 111122ac..5f294761 100644
--- a/lxc/storage.go
+++ b/lxc/storage.go
@@ -772,6 +772,10 @@ func (c *storageCmd) doStoragePoolVolumesList(conf *config.Config, remote string
 		return err
 	}
 
+	if pool == "" {
+		return errArgs
+	}
+
 	volumes, err := client.GetStoragePoolVolumes(pool)
 	if err != nil {
 		return err
