From f6be2f40c7f934099d734099211f510d11d58792 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sat, 20 Jan 2018 22:58:38 +0200
Subject: shared/hostpath: Properly handle prefix check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #4181

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 shared/util.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shared/util.go b/shared/util.go
index 08884bf6..814cca5c 100644
--- a/shared/util.go
+++ b/shared/util.go
@@ -139,7 +139,7 @@ func HostPath(path string) string {
 
 	// Check if the path is already snap-aware
 	for _, prefix := range []string{"/dev", "/snap", "/var/snap", "/var/lib/snapd"} {
-		if strings.HasPrefix(path, prefix) {
+		if path == prefix || strings.HasPrefix(path, fmt.Sprintf("%s/", prefix)) {
 			return path
 		}
 	}
