From 24406036e232f7f30a451cd6714e34c8ae1a4aff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 5 Apr 2018 15:16:06 -0400
Subject: Make new gofmt happy
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/db/migration_test.go | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/lxd/db/migration_test.go b/lxd/db/migration_test.go
index fe7da56d..72469a8b 100644
--- a/lxd/db/migration_test.go
+++ b/lxd/db/migration_test.go
@@ -105,13 +105,7 @@ func TestImportPreClusteringData(t *testing.T) {
 		// The other config keys are node-specific.
 		config, err = query.SelectConfig(tx.Tx(), "storage_pools_config", "node_id=?", 1)
 		require.NoError(t, err)
-		assert.Equal(t,
-			map[string]string{
-				"source": "/foo/bar",
-				"size":   "123",
-				"volatile.initial_source": "/foo/bar",
-				"zfs.pool_name":           "mypool",
-			}, config)
+		assert.Equal(t, map[string]string{"source": "/foo/bar", "size": "123", "volatile.initial_source": "/foo/bar", "zfs.pool_name": "mypool"}, config)
 
 		// Storage volumes have now a node_id key set to 1 (the ID of
 		// the default node).
