From 4458bad5f08d29f6239dfd5bcd174b323f85247c Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 12 Oct 2017 20:19:18 +0200
Subject: dir: make sure pool is mounted

Closes #3938.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/storage_dir.go | 70 ++++++++++++++++++++++++++++++++++++++++++++++--------
 1 file changed, 60 insertions(+), 10 deletions(-)

diff --git a/lxd/storage_dir.go b/lxd/storage_dir.go
index acdbca41..cc39b528 100644
--- a/lxd/storage_dir.go
+++ b/lxd/storage_dir.go
@@ -298,13 +298,18 @@ func (s *storageDir) StoragePoolUpdate(writable *api.StoragePoolPut, changedConf
 func (s *storageDir) StoragePoolVolumeCreate() error {
 	logger.Infof("Creating DIR storage volume \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	source := s.pool.Config["source"]
 	if source == "" {
 		return fmt.Errorf("no \"source\" property found for the storage pool")
 	}
 
 	storageVolumePath := getStoragePoolVolumeMountPoint(s.pool.Name, s.volume.Name)
-	err := os.MkdirAll(storageVolumePath, 0711)
+	err = os.MkdirAll(storageVolumePath, 0711)
 	if err != nil {
 		return err
 	}
@@ -367,13 +372,18 @@ func (s *storageDir) ContainerStorageReady(name string) bool {
 func (s *storageDir) ContainerCreate(container container) error {
 	logger.Debugf("Creating empty DIR storage volume for container \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	source := s.pool.Config["source"]
 	if source == "" {
 		return fmt.Errorf("no \"source\" property found for the storage pool")
 	}
 
 	containerMntPoint := getContainerMountPoint(s.pool.Name, container.Name())
-	err := createContainerMountpoint(containerMntPoint, container.Path(), container.IsPrivileged())
+	err = createContainerMountpoint(containerMntPoint, container.Path(), container.IsPrivileged())
 	if err != nil {
 		return err
 	}
@@ -399,6 +409,11 @@ func (s *storageDir) ContainerCreate(container container) error {
 func (s *storageDir) ContainerCreateFromImage(container container, imageFingerprint string) error {
 	logger.Debugf("Creating DIR storage volume for container \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	source := s.pool.Config["source"]
 	if source == "" {
 		return fmt.Errorf("no \"source\" property found for the storage pool")
@@ -407,7 +422,7 @@ func (s *storageDir) ContainerCreateFromImage(container container, imageFingerpr
 	privileged := container.IsPrivileged()
 	containerName := container.Name()
 	containerMntPoint := getContainerMountPoint(s.pool.Name, containerName)
-	err := createContainerMountpoint(containerMntPoint, container.Path(), privileged)
+	err = createContainerMountpoint(containerMntPoint, container.Path(), privileged)
 	if err != nil {
 		return err
 	}
@@ -455,6 +470,11 @@ func (s *storageDir) ContainerDelete(container container) error {
 		return fmt.Errorf("no \"source\" property found for the storage pool")
 	}
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	// Delete the container on its storage pool:
 	// ${POOL}/containers/<container_name>
 	containerName := container.Name()
@@ -470,7 +490,7 @@ func (s *storageDir) ContainerDelete(container container) error {
 		}
 	}
 
-	err := deleteContainerMountpoint(containerMntPoint, container.Path(), s.GetStorageTypeName())
+	err = deleteContainerMountpoint(containerMntPoint, container.Path(), s.GetStorageTypeName())
 	if err != nil {
 		return err
 	}
@@ -556,6 +576,11 @@ func (s *storageDir) copySnapshot(target container, source container) error {
 func (s *storageDir) ContainerCopy(target container, source container, containerOnly bool) error {
 	logger.Debugf("Copying DIR container storage %s -> %s.", source.Name(), target.Name())
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	ourStart, err := source.StorageStart()
 	if err != nil {
 		return err
@@ -614,7 +639,7 @@ func (s *storageDir) ContainerCopy(target container, source container, container
 }
 
 func (s *storageDir) ContainerMount(c container) (bool, error) {
-	return true, nil
+	return s.StoragePoolMount()
 }
 
 func (s *storageDir) ContainerUmount(name string, path string) (bool, error) {
@@ -624,6 +649,11 @@ func (s *storageDir) ContainerUmount(name string, path string) (bool, error) {
 func (s *storageDir) ContainerRename(container container, newName string) error {
 	logger.Debugf("Renaming DIR storage volume for container \"%s\" from %s -> %s.", s.volume.Name, s.volume.Name, newName)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	source := s.pool.Config["source"]
 	if source == "" {
 		return fmt.Errorf("no \"source\" property found for the storage pool")
@@ -633,7 +663,7 @@ func (s *storageDir) ContainerRename(container container, newName string) error
 	oldContainerSymlink := shared.VarPath("containers", container.Name())
 	newContainerMntPoint := getContainerMountPoint(s.pool.Name, newName)
 	newContainerSymlink := shared.VarPath("containers", newName)
-	err := renameContainerMountpoint(oldContainerMntPoint, oldContainerSymlink, newContainerMntPoint, newContainerSymlink)
+	err = renameContainerMountpoint(oldContainerMntPoint, oldContainerSymlink, newContainerMntPoint, newContainerSymlink)
 	if err != nil {
 		return err
 	}
@@ -674,6 +704,11 @@ func (s *storageDir) ContainerRename(container container, newName string) error
 func (s *storageDir) ContainerRestore(container container, sourceContainer container) error {
 	logger.Debugf("Restoring DIR storage volume for container \"%s\" from %s -> %s.", s.volume.Name, sourceContainer.Name(), container.Name())
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	targetPath := container.Path()
 	sourcePath := sourceContainer.Path()
 
@@ -700,10 +735,15 @@ func (s *storageDir) ContainerGetUsage(container container) (int64, error) {
 func (s *storageDir) ContainerSnapshotCreate(snapshotContainer container, sourceContainer container) error {
 	logger.Debugf("Creating DIR storage volume for snapshot \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	// Create the path for the snapshot.
 	targetContainerName := snapshotContainer.Name()
 	targetContainerMntPoint := getSnapshotMountPoint(s.pool.Name, targetContainerName)
-	err := os.MkdirAll(targetContainerMntPoint, 0711)
+	err = os.MkdirAll(targetContainerMntPoint, 0711)
 	if err != nil {
 		return err
 	}
@@ -772,10 +812,15 @@ onSuccess:
 func (s *storageDir) ContainerSnapshotCreateEmpty(snapshotContainer container) error {
 	logger.Debugf("Creating empty DIR storage volume for snapshot \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	// Create the path for the snapshot.
 	targetContainerName := snapshotContainer.Name()
 	targetContainerMntPoint := getSnapshotMountPoint(s.pool.Name, targetContainerName)
-	err := os.MkdirAll(targetContainerMntPoint, 0711)
+	err = os.MkdirAll(targetContainerMntPoint, 0711)
 	if err != nil {
 		return err
 	}
@@ -857,11 +902,16 @@ func (s *storageDir) ContainerSnapshotDelete(snapshotContainer container) error
 func (s *storageDir) ContainerSnapshotRename(snapshotContainer container, newName string) error {
 	logger.Debugf("Renaming DIR storage volume for snapshot \"%s\" from %s -> %s.", s.volume.Name, s.volume.Name, newName)
 
+	_, err := s.StoragePoolMount()
+	if err != nil {
+		return err
+	}
+
 	// Rename the mountpoint for the snapshot:
 	// ${POOL}/snapshots/<old_snapshot_name> to ${POOL}/snapshots/<new_snapshot_name>
 	oldSnapshotMntPoint := getSnapshotMountPoint(s.pool.Name, snapshotContainer.Name())
 	newSnapshotMntPoint := getSnapshotMountPoint(s.pool.Name, newName)
-	err := os.Rename(oldSnapshotMntPoint, newSnapshotMntPoint)
+	err = os.Rename(oldSnapshotMntPoint, newSnapshotMntPoint)
 	if err != nil {
 		return err
 	}
@@ -871,7 +921,7 @@ func (s *storageDir) ContainerSnapshotRename(snapshotContainer container, newNam
 }
 
 func (s *storageDir) ContainerSnapshotStart(container container) (bool, error) {
-	return true, nil
+	return s.StoragePoolMount()
 }
 
 func (s *storageDir) ContainerSnapshotStop(container container) (bool, error) {
