From e4706f5e3c635acbc7197210d92fd219ab47121a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 9 Apr 2018 02:28:08 +0200
Subject: lxc/alias: Fix example in help message
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #4424

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxc/alias.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxc/alias.go b/lxc/alias.go
index f268924c..bfbc0203 100644
--- a/lxc/alias.go
+++ b/lxc/alias.go
@@ -55,7 +55,7 @@ func (c *cmdAliasAdd) Command() *cobra.Command {
 	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
 		`Add new aliases`))
 	cmd.Example = cli.FormatSection("", i18n.G(
-		`lxc alias add list list -c ns46S
+		`lxc alias add list "list -c ns46S"
     Overwrite the "list" command to pass -c ns46S.`))
 
 	cmd.RunE = c.Run
