From 821f709cc6edf54fd64211e97158f04a5f2deb00 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 10 Apr 2018 14:52:48 +0200
Subject: storage: createContainerMountpoint() fix perms

s/0755/0711/g

Closes #4433.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/storage.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxd/storage.go b/lxd/storage.go
index 5f9f7088..f718027e 100644
--- a/lxd/storage.go
+++ b/lxd/storage.go
@@ -585,7 +585,7 @@ func createContainerMountpoint(mountPoint string, mountPointSymlink string, priv
 	if privileged {
 		mode = 0700
 	} else {
-		mode = 0755
+		mode = 0711
 	}
 
 	mntPointSymlinkExist := shared.PathExists(mountPointSymlink)
@@ -593,7 +593,7 @@ func createContainerMountpoint(mountPoint string, mountPointSymlink string, priv
 
 	var err error
 	if !mntPointSymlinkTargetExist {
-		err = os.MkdirAll(mountPoint, 0755)
+		err = os.MkdirAll(mountPoint, 0711)
 		if err != nil {
 			return err
 		}
