From 5c2c4f01968e0924c8895b8abff1c44432dbb968 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 10 Apr 2018 14:53:38 +0200
Subject: ceph: s/0755/0711/g

Closes #4433.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/storage_ceph.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lxd/storage_ceph.go b/lxd/storage_ceph.go
index 4bd9832e..1839c086 100644
--- a/lxd/storage_ceph.go
+++ b/lxd/storage_ceph.go
@@ -1096,15 +1096,15 @@ func (s *storageCeph) ContainerCreateFromImage(container container, fingerprint
 		}
 		logger.Debugf(`Shifted rootfs for container "%s"`, containerName)
 
-		err = os.Chmod(containerPoolVolumeMntPoint, 0755)
+		err = os.Chmod(containerPoolVolumeMntPoint, 0711)
 		if err != nil {
 			logger.Errorf(`Failed change mountpoint "%s" `+
-				`permissions to 0755 for container "%s" for `+
+				`permissions to 0711 for container "%s" for `+
 				`RBD storage volume: %s`,
 				containerPoolVolumeMntPoint, containerName, err)
 			return err
 		}
-		logger.Debugf(`Changed mountpoint "%s" permissions to 0755 for `+
+		logger.Debugf(`Changed mountpoint "%s" permissions to 0711 for `+
 			`container "%s" for RBD storage volume`,
 			containerPoolVolumeMntPoint, containerName)
 	} else {
