From 90367cb31078d2d0d378b62d9598ec3bd76ca92c Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 10 Apr 2018 14:54:14 +0200
Subject: lvm: s/0755/0711/g

Closes #4433.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/storage_lvm.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lxd/storage_lvm.go b/lxd/storage_lvm.go
index 341ebe3e..df5c7cb3 100644
--- a/lxd/storage_lvm.go
+++ b/lxd/storage_lvm.go
@@ -898,7 +898,7 @@ func (s *storageLvm) ContainerCreate(container container) error {
 		sourceName, _, _ := containerGetParentAndSnapshotName(containerName)
 		snapshotMntPointSymlinkTarget := shared.VarPath("storage-pools", s.pool.Name, "snapshots", sourceName)
 		snapshotMntPointSymlink := shared.VarPath("snapshots", sourceName)
-		err := os.MkdirAll(containerMntPoint, 0755)
+		err := os.MkdirAll(containerMntPoint, 0711)
 		if err != nil {
 			return err
 		}
@@ -909,7 +909,7 @@ func (s *storageLvm) ContainerCreate(container container) error {
 	} else {
 		containerMntPoint := getContainerMountPoint(s.pool.Name, containerName)
 		containerPath := container.Path()
-		err := os.MkdirAll(containerMntPoint, 0755)
+		err := os.MkdirAll(containerMntPoint, 0711)
 		if err != nil {
 			return err
 		}
@@ -955,7 +955,7 @@ func (s *storageLvm) ContainerCreateFromImage(container container, fingerprint s
 
 	containerMntPoint := getContainerMountPoint(s.pool.Name, containerName)
 	containerPath := container.Path()
-	err = os.MkdirAll(containerMntPoint, 0755)
+	err = os.MkdirAll(containerMntPoint, 0711)
 	if err != nil {
 		return err
 	}
@@ -985,7 +985,7 @@ func (s *storageLvm) ContainerCreateFromImage(container container, fingerprint s
 	if container.IsPrivileged() {
 		err = os.Chmod(containerMntPoint, 0700)
 	} else {
-		err = os.Chmod(containerMntPoint, 0755)
+		err = os.Chmod(containerMntPoint, 0711)
 	}
 	if err != nil {
 		return err
