From 665ef4d26752fbbb1633d2ba779da9194661884b Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 10 Apr 2018 14:55:56 +0200
Subject: zfs: s/0755/0711/g

Closes #4433.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/storage_zfs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/storage_zfs.go b/lxd/storage_zfs.go
index e657c308..e3a0594d 100644
--- a/lxd/storage_zfs.go
+++ b/lxd/storage_zfs.go
@@ -129,7 +129,7 @@ func (s *storageZfs) StoragePoolCreate() error {
 	}()
 
 	storagePoolMntPoint := getStoragePoolMountPoint(s.pool.Name)
-	err = os.MkdirAll(storagePoolMntPoint, 0755)
+	err = os.MkdirAll(storagePoolMntPoint, 0711)
 	if err != nil {
 		return err
 	}
