From b1cf8d1ab8f0b2fc6592d33912b4007a4aa365e7 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 11 Apr 2018 14:14:34 +0200
Subject: sys/fs: s/MkdirAll/Mkdir/g

Closes #4433.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/sys/fs.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lxd/sys/fs.go b/lxd/sys/fs.go
index c8550fc3..8370838e 100644
--- a/lxd/sys/fs.go
+++ b/lxd/sys/fs.go
@@ -26,10 +26,13 @@ func (s *OS) initDirs() error {
 		{filepath.Join(s.VarDir, "disks"), 0700},
 		{filepath.Join(s.VarDir, "storage-pools"), 0711},
 	}
+
 	for _, dir := range dirs {
-		if err := os.MkdirAll(dir.path, dir.mode); err != nil {
+		err := os.Mkdir(dir.path, dir.mode)
+		if err != nil && !os.IsExist(err) {
 			return err
 		}
 	}
+
 	return nil
 }
